lib/rubocop/file_finder.rb

```diff
@@ -31,17 +31,26 @@ def find_files_upwards(filename, start_dir, use_home: false)
     private
 
     def traverse_files_upwards(filename, start_dir, use_home)
+      traverse_file(filename, start_dir).each { |f| yield(f.to_s) }
+      return unless use_home && (ENV.key?('HOME') ||
```

Comment:
Seems to me that we're mixing different concerns in this method. Ideally the file traversal should be separate from checks for which directory contains the config file.